orbstation/antagonists/wizard_journeyman/diploma/spell_defensive.dm

```diff
@@ -0,0 +1,138 @@
+/**
+ * Spells which help you survive or run away.
+ * Includes almost all transit spells, which you really REALLY want at least one of.
+ */
+
+/datum/diploma_spell/disabletech
+	name = "Disable Tech"
+	desc = "Disables all weapons, cameras and most other technology in range."
+	spell_type = /datum/action/cooldown/spell/emp/disable_tech
+	category = DIPLOMA_SPELL_DEFENSIVE
+
+/datum/diploma_spell/repulse
+	name = "Repulse"
+	desc = "Throws everything around the user away."
+	spell_type = /datum/action/cooldown/spell/aoe/repulse/wizard
+	category = DIPLOMA_SPELL_DEFENSIVE
+
+/datum/diploma_spell/lichdom
+	name = "Lichdom"
+	desc = "Bind your soul to an item on the space station to become an immortal skeleton. \
+		So long as that item remains intact, you cannot be permanently killed. \
+		Be warned that the crew may be able to track your spirit back to where you stashed your soulstone."
+	spell_type = /datum/action/cooldown/spell/lichdom
+	category = DIPLOMA_SPELL_DEFENSIVE
+	weight = DIPLOMA_SPELL_RARE
+
+/datum/diploma_spell/timestop
+	name = "Time Stop"
+	desc = "Stops time for everyone except for you, allowing you to move freely \
+		while your enemies and even projectiles are frozen."
+	spell_type = /datum/action/cooldown/spell/timestop
+	category = DIPLOMA_SPELL_DEFENSIVE
+	weight = DIPLOMA_SPELL_RARE
+
+/datum/diploma_spell/smoke
+	name = "Smoke"
+	desc = "Spawns a cloud of choking smoke at your location."
+	spell_type = /datum/action/cooldown/spell/smoke
+	category = DIPLOMA_SPELL_DEFENSIVE
+
+/datum/diploma_spell/forcewall
+	name = "Force Wall"
+	desc = "Create a magical barrier that only you can pass through."
+	spell_type = /datum/action/cooldown/spell/forcewall
+	category = DIPLOMA_SPELL_DEFENSIVE
+
+/datum/diploma_spell/spacetime_dist
+	name = "Spacetime Distortion"
+	desc = "Entangle the strings of space-time in an area around you, \
+		randomizing the layout and making proper movement impossible. The strings vibrate..."
+	spell_type = /datum/action/cooldown/spell/spacetime_dist
+	category = DIPLOMA_SPELL_DEFENSIVE
+
+/datum/diploma_spell/item/lockerstaff
+	name = "Staff of the Locker"
+	desc = "A staff that shoots lockers. It eats anyone it hits on its way, leaving a welded locker with your victims behind. \
+		Remember that any fool can use a stave dropped onto the ground."
+	item_path = /obj/item/gun/magic/staff/locker
+	category = DIPLOMA_SPELL_DEFENSIVE
+
+/datum/diploma_spell/knock
+	name = "Knock"
+	desc = "Opens nearby doors and closets."
+	spell_type = /datum/action/cooldown/spell/aoe/knock
+	category = DIPLOMA_SPELL_DEFENSIVE
+
+/datum/diploma_spell/blink
+	name = "Blink"
+	desc = "Randomly teleports you a short distance."
+	spell_type = /datum/action/cooldown/spell/teleport/radius_turf/blink
+	category = DIPLOMA_SPELL_DEFENSIVE
+
+/datum/diploma_spell/mirror_walk
```

Comment:
This is a variant of the spell that the Maid in the Mirror has, right? I like the idea of a wizard being able to use that spell, but I would note that as-is, it allows you to travel to most locations on the station (mostly aside from maintenance) completely intangibly and invisibly at very high speeds, and also it's possible to hide in the mirror realm for as long as you want, which seems pretty strong compared to the other jaunt spells. Could maybe use some sort of drawback.

Reply:
Yeah this one might be a bit too easy, I may try to add asphyxiation damage while it is active or something like that.

Reply:
While doing this I have decided that I hate jaunt code and need to refactor it upstream, the icon for this spell is messed up for this reason but I won't fix it here because I'd be repeating the work later.

Reply:
ok after a lot of time spent on this I think I cannot add it until jaunting is refactored so I will push what I have and turn off the spell until then

Reply:
As always I figured out a way just after giving up, it will be able to be cleaner after a Jaunt refactor though.